hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusDeleteTopicHandler.java

```diff
@@ -51,25 +53,26 @@ public ConsensusDeleteTopicHandler() {
         // Exists for injection
     }
 
+    @Override
+    public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckException {
+        final ConsensusDeleteTopicTransactionBody op = txn.consensusDeleteTopicOrThrow();
+        validateTruePreCheck(op.hasTopicID(), INVALID_TOPIC_ID);
```

Comment:
This checks that the object exists, but does not check if the topic ID is valid. We may as well check the shard num, realm num, and topic num here too

Reply:
This was moved from preHandle(). Not sure if we check shard, real and entity number in pureChecks(). It will fail in preHandle if they are invalid since topic will be null.

Reply:
It will fail in preHandle, true. Wasn't the point to move anything we can into `pureChecks`?

Reply:
We currently don't check shard, realm. I checked all other handlers' pureChecks(), entity ID number like topic num, file num, account num, etc. are not checked in pureChecks(). Should we add this check everywhere?

Reply:
Mono doesn't check these, so we don't need to add them right now.